receiver/oracledbreceiver/scraper.go

```diff
@@ -579,10 +716,230 @@ func (s *oracleScraper) scrapeLogs(ctx context.Context) (plog.Logs, error) {
+				sqlId:        row[sqlIDAttr],
+				queryText:    row[sqlTextAttr],
+				childNumber:  row[childNumberAttr],
+				childAddress: row[childAddressAttr],
+				metrics:      make(map[string]int64, len(enabledColumns)),
+			}
+
+			// it is possible we get a record with all deltas equal to zero. we don't want to process it any further
+			var possiblePurge, positiveDelta bool
+			for _, columnName := range enabledColumns {
+				delta := newCacheVal[columnName] - oldCacheVal[columnName]
+
+				// if any of the deltas is less than zero, cursor was likely purged from the shared pool
+				if delta < 0 {
+					possiblePurge = true
+					break
+				} else if delta > 0 {
+					positiveDelta = true
+				}
+
+				hit.metrics[columnName] = delta
+			}
+
+			// skip if possible purge or all the deltas are equal to zero
+			if !possiblePurge && positiveDelta {
+				hits = append(hits, hit)
+			} else {
+				discardedHits++
+			}
+		}
+		s.metricCache.Add(cacheKey, newCacheVal)
+		cacheUpdates++
+	}
+
+	// if cache updates is not equal to rows returned, that indicates there is problem somewhere
+	s.logger.Info("Cache update", zap.Int("update-count", cacheUpdates), zap.Int("new-size", s.metricCache.Len()))
+
+	if len(hits) == 0 {
+		s.logger.Info("No log records for this scrape")
+		return errors.Join(errs...)
+	}
+
+	s.logger.Info("Cache hits", zap.Int("hit-count", len(hits)), zap.Int("discarded-hit-count", discardedHits))
+	for _, hit := range hits {
+		s.logger.Debug(fmt.Sprintf("Cache hit, SQL_ID: %v, CHILD_NUMBER: %v", hit.sqlId, hit.childNumber), zap.Int64("elapsed-time", hit.metrics[elapsedTimeMetric]))
+	}
+
+	// order by elapsed time delta, descending
+	sort.Slice(hits, func(i, j int) bool {
+		return hits[i].metrics[elapsedTimeMetric] > hits[j].metrics[elapsedTimeMetric]
+	})
+
+	for _, hit := range hits {
+		s.logger.Debug(fmt.Sprintf("Cache hit after sorting, SQL_ID: %v, CHILD_NUMBER: %v", hit.sqlId, hit.childNumber), zap.String("child-address", hit.childAddress), zap.Int64("elapsed-time", hit.metrics[elapsedTimeMetric]))
+	}
+
+	// keep at most maxHitSize
+	hitCountBefore := len(hits)
+	maxHitsSize := min(len(hits), int(s.topQueryCollectCfg.TopQueryCount))
+	hits = hits[:maxHitsSize]
+	skippedCacheHits := hitCountBefore - len(hits)
+	s.logger.Info("Skipped cache hits", zap.Int("count", skippedCacheHits))
+
+	for _, hit := range hits {
+		s.logger.Debug(fmt.Sprintf("Final cache hit, SQL_ID: %v, CHILD_NUMBER: %v", hit.sqlId, hit.childNumber), zap.String("child-address", hit.childAddress), zap.Any("metrics", hit.metrics))
+	}
+	hits = s.obfuscateCacheHits(hits)
+	childAddressToPlanMap := s.getChildAddressToPlanMap(ctx, hits)
+
+	topNResourceLog := logs.ResourceLogs().AppendEmpty()
+
+	resourceAttributes := topNResourceLog.Resource().Attributes()
+	if s.metricsBuilderConfig.ResourceAttributes.OracledbInstanceName.Enabled {
+		resourceAttributes.PutStr("instance.name", s.instanceName)
+	}
+	resourceAttributes.PutStr("db.system.name", "oracle.db")
```

Comment:
Please check with Sam. He mentioned that db.system.name is not a resource attribute. It supposed to go into attributes.